File: app/src/main/java/com/machiav3lli/derdiedas/utils/DatabaseUtil.java
package com.machiav3lli.derdiedas.utils;

import android.content.Context;

import com.machiav3lli.derdiedas.data.Noun;
import com.machiav3lli.derdiedas.data.NounDao;
import com.machiav3lli.derdiedas.data.NounDatabase;

import java.util.List;

public class DatabaseUtil {
    private final NounDao database;

    public DatabaseUtil(Context context) {
        database = NounDatabase.Companion.getInstance(context).getNounDao();
    }

    public void addAllNouns(List<Noun> nouns) {
        database.deleteAll();
        database.insert(nouns);
    }

    public List<Noun> getAllNouns() {
        return database.getAll();
    }

    public long getNounsCount() {
        return database.count();
    }
}

File: app/src/main/java/com/machiav3lli/derdiedas/utils/DatabaseUtil.kt
package com.machiav3lli.derdiedas.utils

import android.content.Context
import com.machiav3lli.derdiedas.data.Noun
import com.machiav3lli.derdiedas.data.NounDatabase
import java.io.UnsupportedEncodingException

fun Context.createNounListFromAsset(): MutableList<Noun> {
    var nounsString: String? = null
    try {
        nounsString = FileUtils.getNounList(this)
    } catch (e: UnsupportedEncodingException) {
        e.printStackTrace()
    }
    val nouns = FileUtils.getLines(nounsString!!)
    return nouns.map {
        val noun = it.split(",").toTypedArray()[0]
        val gender = it.split(",").toTypedArray()[1]
        Noun(noun, gender, 0)
    }.toMutableList()
}

fun Context.getNounsCount(): Long {
    val db = NounDatabase.getInstance(this)
    return db.nounDao.count
}
